compensated-ruby/lib/compensated/gumroad/event_parser.rb

```diff
@@ -1,14 +1,18 @@
 require "webrick"
 module Compensated
   module Gumroad
-    class EventParser
+    class EventParser < Compensated::EventParser
       def parses?(request)
         # Gumroad pings are always form data
         return false unless request.form_data?
         keys = request.data.keys.map(&:to_sym)
         keys.include?(:seller_id) && keys.include?(:product_id) && keys.include?(:product_permalink) && request.data["product_permalink"].include?("gum.co")
       end
 
+      # Transform Gumroad input data into Compensated event hash
+      #
+      # @param input [String, Rack::Request]
```

Comment:
I didn't put `IO` here because I don't see it if it takes `IO` like the others as it doesn't use `Compensated.json_adapter`, but I also wasn't 100% confident.